### PR TITLE
1198-dev-wrap-accessagentcaller-into-jpsagent

### PR DESCRIPTION
**Pull request by csl37**
Wrapped the query and update methods from the AccessAgentCaller class into JPSAgent and also added the methods to JPSAgentInterface. This allows agents extending the JPSAgent class to call the AccessAgent to execute SPARQL query and update operations on the KG.

### CODE REVIEW

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java [hunk 1]
```diff
@@ -54,5 +55,34 @@ public boolean validateInput(JSONObject requestParams) throws BadRequestExceptio
         }
         return true;
     }
-
+    
+    /**
+     * Execute a <a href="https://www.w3.org/TR/sparql11-query/">SPARQL Query</a> on the target resource
```
Comment (achCARES) [inline JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java:60]: You can use 

`{@link <a href="https://www.w3.org/TR/sparql11-query/">}` 

in code comments for those things.

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java [hunk 2]
```diff
@@ -54,5 +55,34 @@ public boolean validateInput(JSONObject requestParams) throws BadRequestExceptio
...
+     * 							e.g. to access the Ontokin triple store
+     * 							both "ontokin" and "http://www.theworldavatar.com/kb/ontokin" are accepted.
+     * @param sparqlQuery		SPARQL query string
+     * @return 	the query result in the W3C Query result JSON format, 
+     * see <a href="https://www.w3.org/TR/sparql11-results-json/">Query Results JSON Format</a>
+     */
+    public String query(String targetResourceID, String sparqlQuery) {
+    	//pass the target resource ID directly as the targetUrl
+    	//both datasetUrl and targetUrl are not used by the AccessAgent for queries
+    	return AccessAgentCaller.query(null, targetResourceID, sparqlQuery);
```
Comment (achCARES) [inline JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java:72]: Maybe we could just have a second method for that in the `AccessAgentCaller` that does not need to pass the dataset/thatgetURL in the signature: like `query(String targetResourceID, String sparqlQuery)`.

**Review by achCARES — CHANGES_REQUESTED** [review]

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java (hunk 1 above)
Comment (csl37) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java]: Changed

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java (hunk 2 above)
Comment (csl37) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java]: Added query(String targetResourceID, String sparqlQuery) and update(String targetResourceID, String sparqlUpdate) to AccessAgentCaller

**Review by achCARES — APPROVED** [review]
✅

**Pull request merged**